### PR TITLE
Plat 606 reverse proxy ofelia refactor

### PR DESCRIPTION
**Pull request by nour-borgi**

### CODE REVIEW

**Review by IlyesBenAmara — APPROVED** [review]

**Review by marrouchi — APPROVED** [review]

**Review by marrouchi — APPROVED** [review]

File: reverse-proxy-nginx/swarm.sh [hunk 1]
```diff
@@ -1,198 +1,166 @@
...
 
-      log info "Updating nginx service: adding config file..."
-      try "docker service update --config-add source=${TIMESTAMPED_NGINX},target=/etc/nginx/nginx.conf instant_reverse-proxy-nginx" "Error updating nginx service"
+  config::generate_service_configs "$service_name" /etc/nginx/conf.d "${COMPOSE_FILE_PATH}/package-conf-${DEPLOY_TYPE}" "${COMPOSE_FILE_PATH}" "nginx"
+  nginx_temp_compose_param="docker-compose.tmp.yml"
 
-      log info "Done updating nginx service"
-    else
-      log info "Running reverse-proxy package in SECURE mode"
+  docker::deploy_service "${COMPOSE_FILE_PATH}" "docker-compose.yml" "$nginx_temp_compose_param"
```
Comment (Mark-Labuschagne) [inline reverse-proxy-nginx/swarm.sh:83]:
```suggestion
  docker::deploy_service "${COMPOSE_FILE_PATH}" "docker-compose.yml" "docker-compose.tmp.yml"
```

File: reverse-proxy-nginx/swarm.sh [hunk 2]
```diff
@@ -1,198 +1,166 @@
...
+}
 
-      try "docker config create --label name=nginx ${TIMESTAMPED_NGINX} ${COMPOSE_FILE_PATH}/config/nginx-temp-insecure.conf" "Failed to create nginx insecure config"
+function deploy_nginx() {
+  local -r DEPLOY_TYPE=${1:?"FATAL: deploy_nginx DEPLOY_TYPE not provided"}
 
-      log info "Updating nginx service: adding config file..."
-      try "docker service update --config-add source=${TIMESTAMPED_NGINX},target=/etc/nginx/nginx.conf instant_reverse-proxy-nginx" "Error updating nginx service"
+  config::generate_service_configs "$service_name" /etc/nginx/conf.d "${COMPOSE_FILE_PATH}/package-conf-${DEPLOY_TYPE}" "${COMPOSE_FILE_PATH}" "nginx"
+  nginx_temp_compose_param="docker-compose.tmp.yml"
```
Comment (Mark-Labuschagne) [inline reverse-proxy-nginx/swarm.sh:81]:
```suggestion
```

File: reverse-proxy-nginx/swarm.sh [hunk 3]
```diff
@@ -1,198 +1,166 @@
...
+    log info "Running Nginx Reverse Proxy package in INSECURE mode"
+    (
+      deploy_nginx "insecure"
 
-      local domain_args=()
-      if [ -n "$SUBDOMAINS" ]; then
-        domain_args=(-d "${DOMAIN_NAME},${SUBDOMAINS}")
-      else
-        domain_args=(-d "${DOMAIN_NAME}")
+      if [ "${INSECURE_PORTS}" != "" ]; then
```
Comment (Mark-Labuschagne) [inline reverse-proxy-nginx/swarm.sh:93]:
```suggestion
      if [[ "${INSECURE_PORTS}" != "" ]]; then
```

File: reverse-proxy-nginx/swarm.sh [hunk 4]
```diff
@@ -1,198 +1,166 @@
...
 
-      config::generate_service_configs reverse-proxy-nginx /etc/nginx/conf.d "${COMPOSE_FILE_PATH}/package-conf-insecure" "${COMPOSE_FILE_PATH}" nginx
-      nginx_temp_compose_param="-c ${COMPOSE_FILE_PATH}/docker-compose.tmp.yml"
-      try "docker stack deploy -c ${COMPOSE_FILE_PATH}/docker-compose.yml $nginx_temp_compose_param instant" "Failed to deploy nginx"
+function publish_insecure_ports() {
+  IFS='-' read -ra PORTS <<<"$INSECURE_PORTS"
+  local ports_array=()
+  for i in "${PORTS[@]}"; do
+    IFS=':' read -ra PORTS_SPLIT <<<"$i"
+    if [ "${PORTS_SPLIT[0]}" != "" ] && [ "${PORTS_SPLIT[1]}" != "" ]; then
```
Comment (Mark-Labuschagne) [inline reverse-proxy-nginx/swarm.sh:48]:
```suggestion
    if [[ "${PORTS_SPLIT[0]}" != "" ]] && [[ "${PORTS_SPLIT[1]}" != "" ]]; then
```

File: reverse-proxy-nginx/swarm.sh [hunk 5]
```diff
@@ -1,198 +1,166 @@
...
+function import_sources() {
+  source "${UTILS_PATH}/docker-utils.sh"
+  source "${UTILS_PATH}/config-utils.sh"
+  source "${UTILS_PATH}/log.sh"
+}
 
-      config::generate_service_configs reverse-proxy-nginx /etc/nginx/conf.d "${COMPOSE_FILE_PATH}/package-conf-insecure" "${COMPOSE_FILE_PATH}" nginx
-      nginx_temp_compose_param="-c ${COMPOSE_FILE_PATH}/docker-compose.tmp.yml"
-      try "docker stack deploy -c ${COMPOSE_FILE_PATH}/docker-compose.yml $nginx_temp_compose_param instant" "Failed to deploy nginx"
+function publish_insecure_ports() {
```
Comment (Mark-Labuschagne) [inline reverse-proxy-nginx/swarm.sh:43]: Please space this out better.

File: reverse-proxy-nginx/set-secure-mode.sh [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+    fi
+}
+
+set_secure_mode() {
+    init_vars "$@"
+    import_sources
+
+    local domain_args=()
+
+    if [ -n "$SUBDOMAINS" ]; then
```
Comment (Mark-Labuschagne) [inline reverse-proxy-nginx/set-secure-mode.sh:159]:
```suggestion
    if [[ -n "$SUBDOMAINS" ]]; then
```

File: reverse-proxy-nginx/set-secure-mode.sh [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+    sed -i "s/domain_name/${DOMAIN_NAME}/g;" "${COMPOSE_FILE_PATH}"/config/nginx.conf
+
+    #Set a docker network for nginx
+    set_nginx_network
+
+    #Update nginx to use the dummy certificates
+    update_nginx_dummy_certificates
+
+    local staging_args=""
+    if [ "${STAGING}" == "true" ]; then
```
Comment (Mark-Labuschagne) [inline reverse-proxy-nginx/set-secure-mode.sh:181]:
```suggestion
    if [[ "${STAGING}" == "true" ]]; then
```

File: reverse-proxy-nginx/set-secure-mode.sh [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+            throw \
+            "Failed to create cert-renewal-network network"
+    fi
+}
+
+set_secure_mode() {
+    init_vars "$@"
+    import_sources
+
+    local domain_args=()
```
Comment (Mark-Labuschagne) [inline reverse-proxy-nginx/set-secure-mode.sh:157]: `domain_args` should be declared and initialised in `init_vars`

**Review by Mark-Labuschagne — CHANGES_REQUESTED** [review]

File: reverse-proxy-nginx/swarm.sh (hunk 5 above)
Comment (nour-borgi) [reply reverse-proxy-nginx/swarm.sh:43]: cool done

File: reverse-proxy-nginx/set-secure-mode.sh (hunk 8 above)
Comment (nour-borgi) [reply reverse-proxy-nginx/set-secure-mode.sh]: nice! thanks done

**Review by Mark-Labuschagne — APPROVED** [review]

**Pull request merged**